server/datastore/mysql/hosts.go

```diff
@@ -1383,13 +1383,17 @@ func (*Datastore) getBatchExecutionFilters(whereParams []interface{}, opt fleet.
 		batchScriptExecutionJoin += ` LEFT JOIN host_script_results hsr ON bsehr.host_execution_id = hsr.execution_id`
 		switch opt.BatchScriptExecutionStatusFilter {
 		case fleet.BatchScriptExecutionRan:
-			batchScriptExecutionFilter += ` AND hsr.exit_code = 0`
+			batchScriptExecutionFilter += ` AND hsr.exit_code = 0 && hsr.canceled = 0`
 		case fleet.BatchScriptExecutionPending:
 			// Pending can mean "waiting for execution" or "waiting for results".
-			batchScriptExecutionJoin += ` LEFT JOIN upcoming_activities ua ON ua.execution_id = bsehr.host_execution_id`
-			batchScriptExecutionFilter += ` AND ((ua.execution_id IS NOT NULL) OR (hsr.host_id is NOT NULL AND hsr.exit_code IS NULL AND hsr.canceled = 0 AND bsehr.error IS NULL))`
+			// hsr.exit_code IS NULL <- this means the script has not reported back
+			// (hsr.canceled IS NULL OR hsr.canceled = 0) <- this can mean the script is running, or that it hasn't been activated yet,
+			//                      but either way we haven't canceled it.
+			// bsehr.error IS NULL <- this means the batch script framework didn't mark this host as incompatible
+			//                        with this script run.
+			batchScriptExecutionFilter += ` AND (hsr.exit_code IS NULL AND (hsr.canceled IS NULL OR hsr.canceled = 0) AND bsehr.error IS NULL)`
 		case fleet.BatchScriptExecutionErrored:
-			batchScriptExecutionFilter += ` AND hsr.exit_code > 0`
+			batchScriptExecutionFilter += ` AND hsr.exit_code > 0 && hsr.canceled = 0`
```

Comment:
Consistency nit:
```suggestion
			batchScriptExecutionFilter += ` AND hsr.exit_code > 0 AND hsr.canceled = 0`
```

Reply:
good catch, not intentional, surprised this works at all